### DISCUSSION TITLE
getIsSomeRowsSelected return false if all the rows selected

### DISCUSSION THREAD

**Discussion by raz-moyal** [TanStack/table#4780] [Q&A]
why is "getIsSomeRowsSelected" return false if all row in the table are selected ?

https://github.com/TanStack/table/blob/b14b7af192e684abc110ef3283e8eafdb55f5234/packages/table-core/src/features/RowSelection.ts#L308

**Comment by anjitpariyar** [comment]
exactly why? im also suprised

**Comment by KevinVandy** [comment]
I want to change this too. But there were other people who wanted it to work this way for their own indeterminate logic.

**Reply by tzelon-cypator** [reply]
is there a good reasoning for this or it is what it is?

**Reply by KevinVandy** [reply]
Not really. I want to fix this. It would technically be a breaking change though

**Comment by smucode** [comment]
Very surprising behavior indeed. Perhaps the API is optimized for the usecase where you want to know if some (but not all) rows are selected. If so, I'd argue it's the special case and should be checking `getIsSomeRowsSelected() && !getIsAllRowsSelected()`. More intuitive if `getIsSomeRowsSelected` follows similar semantics as [Array.some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)

**Comment by gtarsia** [comment]
Might be obvious at this point, but this is the real "any" condition: `table.getIsSomeRowsSelected() || table.getIsAllRowsSelected()`.
